WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SettingsUsernameChangerFragment.kt

```diff
@@ -0,0 +1,193 @@
+            create()
+        }.show().also { alertDialog ->
+            // The change username button is disabled at start.
+            val changeUsernameButton = alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).apply {
+                isEnabled = false
+            }
+
+            // Toggles the enabled property of the button based on the correctness of the username being typed.
+            usernameControl.addTextChangedListener(object : TextWatcher {
+                override fun afterTextChanged(editable: Editable?) {
+                    editable?.let {
+                        changeUsernameButton.isEnabled = (it.toString() == username)
+                    }
+                }
+
+                override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                }
+
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                }
+            })
+            usernameControl.requestFocus()
+        }
+    }
+
+    private fun saveUsername(username: String) {
+        showProgress()
+        val payload = PushUsernamePayload(
+                username, KEEP_OLD_SITE_AND_ADDRESS
+        )
+        mDispatcher.dispatch(AccountActionBuilder.newPushUsernameAction(payload))
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onUsernameChanged(event: OnUsernameChanged) {
+        if (event.isError) {
+            AnalyticsTracker.track(ACCOUNT_SETTINGS_CHANGE_USERNAME_FAILED)
+            AppLog.e(
+                    T.API, "SettingsUsernameChangerFragment.onUsernameChanged: " +
+                    event.error.type + " - " + event.error.message
+            )
+            endProgress()
+            showErrorDialog(SpannableStringBuilder(getString(R.string.signup_epilogue_error_generic)))
+        } else if (event.username != null) {
+            AnalyticsTracker.track(ACCOUNT_SETTINGS_CHANGE_USERNAME_SUCCEEDED)
+            endProgress()
+            val result = Bundle().apply { putString(RESULT_USERNAME, event.username) }
+            dialogController.confirm(result)
+        }
+    }
+
+    fun showProgress() {
+        if (progressDialog == null || progressDialog?.window == null || progressDialog?.isShowing == false) {
+            progressDialog = ProgressDialog(context).apply {
+                isIndeterminate = true
+                setCancelable(true)
+                setMessage(getString(R.string.settings_username_changer_progress_dialog))
+                setOnCancelListener { showActionCancelMessage() }
+            }
+        }
+        try {
+            if (progressDialog?.isShowing == false) {
+                progressDialog?.show()
+            }
+        } catch (e: WindowManager.BadTokenException) {
+            AppLog.e(T.SETTINGS, e)
+        }
+    }
+
+    private fun endProgress() {
+        if (progressDialog != null && progressDialog?.isShowing == true && progressDialog?.window != null) {
```

Comment:
NitPick: I'd consider using 
```
progressDialog?.let {
...
}
```

So we don't need to keep repeating `?`. Wdyt?

Note: Might be worth considering even at some other places.

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/10436/commits/01c8d819703b7a848b1c702ad46b98e59d03ce9e